File: mobile/lib/models/unit.dart
import 'enums.dart';
import 'parse.dart';

class Unit {
  const Unit({
    required this.id,
    required this.kontrakanId,
    required this.name,
    required this.status,
    required this.price,
    required this.createdAt,
  });

  final int id;
  final int kontrakanId;
  final String name;
  final UnitStatus status;
  final double? price;
  final DateTime createdAt;

  factory Unit.fromJson(Map<String, dynamic> json) => Unit(
        id: json['id'] as int,
        kontrakanId: json['kontrakan_id'] as int,
        name: json['name'] as String,
        status: UnitStatus.fromApi(json['status'] as String),
        price: parseNullableDouble(json['price']),
        createdAt: parseDate(json['created_at']),
      );
}

File: backend/app/schemas/unit.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import UnitStatus


class UnitCreate(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    status: UnitStatus = UnitStatus.kosong
    price: Decimal | None = Field(default=None, ge=0)


class UnitUpdate(BaseModel):
    # semua opsional: klien kirim hanya field yang ingin diubah
    name: str | None = Field(default=None, min_length=1, max_length=80)
    status: UnitStatus | None = None
    price: Decimal | None = Field(default=None, ge=0)


class UnitOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    kontrakan_id: int
    name: str
    status: UnitStatus
    price: Decimal | None
    created_at: datetime
